File: expressdine-client/src/Services/OrderService.js

import axios from "axios";
const ed_url = "http://localhost:8080/order";

export class OrderService{
  placeOrder = (userId) => {
    return axios.put(ed_url + "/placeOrder/" + userId);
  };

  getByUser = (userId) => {
    return axios.get(ed_url + "/getByUser/" + userId);
  };
  
  getOrders = (ownerId) => {
    return axios.get(ed_url + "/getOrders/"+ownerId);
  };

  getaccepted = () => {
    return axios.get(ed_url + "/getaccepted");
  };

  acceptOrder = (orderId) => {
    return axios.put(ed_url + "/acceptOrder/" + orderId);
  };
}

export default new OrderService();

File: ExpressDine_Server/src/main/java/com/cybage/service/OrderService.java
package com.cybage.service;

import java.util.List;

import com.cybage.model.Order;

public interface OrderService {

	void placeOrder(int userId);

	List<Order> getByUser(int id);

	void acceptOrder(int orderId);


	List<Order> getOrders(int ownerId);

}
